drivers/pinctrl/pinctrl_imx.c

```diff
@@ -5,7 +5,9 @@
  */
 
 #include <zephyr/drivers/pinctrl.h>
+#ifndef CONFIG_ARM64
 #include <soc.h>
+#endif
 
```

Comment:
This inclusion of soc.h does not seem to be needed in arm32 either so I would completely remove this one. Is there an acceptable tool and process we can go through to identify unused header files?